File: server/src/main/java/com/example/schoolsite/entity/Subject.java
package com.example.schoolsite.entity;

import com.google.firebase.database.annotations.NotNull;
import io.leangen.graphql.annotations.GraphQLQuery;
import lombok.*;

import javax.persistence.*;
import java.util.Objects;

@Entity
@Getter @Setter
@NoArgsConstructor
@ToString @EqualsAndHashCode
public class Subject {
    @Id
    @GeneratedValue //(strategy = GenerationType.AUTO)
    @GraphQLQuery(name="id", description = "A subject`s id")
    private Long id;
    //@Column(name = "subjectName", nullable = false)
    @GraphQLQuery(name="subject_name", description = "A subject`s name")
    private @NotNull  String subjectName;
}

File: server/src/main/java/com/example/schoolsite/entity/Subject.kt
package com.example.schoolsite.entity

import lombok.Getter

@Entity
@Getter
@Setter
@NoArgsConstructor
@Table(name = "subjects")
@ToString
class Subject(@field:Column(name = "subject_name", nullable = false) private val subjectName: String) {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private val id: Long? = null

    @OneToOne(optional = false, mappedBy = "subjectID")
    private val shedule: Shedule? = null

    @Override
    override fun equals(o: Object?): Boolean {
        if (this === o) return true
        if (o == null || getClass() !== o.getClass()) return false
        val subject = o as Subject
        return Objects.equals(id, subject.id) && Objects.equals(subjectName, subject.subjectName)
    }

    @Override
    override fun hashCode(): Int {
        return Objects.hash(id, subjectName)
    }
}